tests/fugue_sql/test_workflow_parse.py

```diff
@@ -365,6 +366,52 @@ def test_select():
     )
 
 
+def test_select_with():
+    dag = FugueWorkflow()
+    dag.select(
+        "with x as ( select * from a ) , y as ( select * from b ) "
+        "select * from x union select * from y"
+    )
+    assert_eq(
+        """
+    with
+        x as (select * from a),
+        y as (select * from b)
+    select *   from x union select * from y
+    
+    """,
+        dag,
+    )
+
+
+def test_select_plus_engine():
+    class MockEngine(SqliteEngine):
+        def __init__(self, execution_engine, p: int = 0):
+            super().__init__(execution_engine)
+            self.p = p
+
+    dag = FugueWorkflow()
+    dag.select("select * from xyz", sql_engine=MockEngine)
+    dag.select("select * from xyz", sql_engine=MockEngine, sql_engine_params={"p": 2})
+
+    temp = dag.select("select a , b from a", sql_engine=MockEngine)
+    temp.transform(mock_transformer2)
+
+    temp = dag.select("select aa , bb from a", sql_engine=MockEngine)
+    dag.select("select aa + bb as t from", temp)
+    assert_eq(
+        """
+    connect MockEngine select * from xyz
+    connect MockEngine(p=2) select * from xyz
+    
+    transform (connect MockEngine select a,b from a) using mock_transformer2
```

Comment:
Feels like this syntax means you can use different Engines for different extensions throughout an end-to-end workflow. Is this the intention? 

But even if it's not the intention, I suppose with the way FugueSQL is structured, there is no way to just fine the Engine once and apply it to everything similar to FugueWorkflow an ExecutionEngine.